spring-ai-2/src/main/java/com/baeldung/springai/nova/AuthorFetcher.java

```diff
@@ -0,0 +1,16 @@
+package com.baeldung.springai.nova;
+
+import java.util.function.Function;
+
+public class AuthorFetcher implements Function<AuthorFetcher.Query, AuthorFetcher.Author> {
+
+    @Override
+    public Author apply(Query author) {
+        return new Author("John Doe", "john.doe@baeldung.com");
+    }
+
+    public record Author(String name, String emailId) { }
+
+    public record Query(String articleTitle) { }
```

Comment:
For consistency with the rest of the record formatting:
```suggestion
    public record Author(String name, String emailId) {
    }

    public record Query(String articleTitle) {
    }
```

Reply:
Feedback incorporated.